### PR TITLE
Add revoke follow feature

### LINKED ISSUE #10754: Offer users to revoke follows besides blocking
It is currently possible to unfollow someone, but not directly to make someone unfollow us. It used to be silently done on contact drop, but [this feature is going away in version 2021.12](https://github.com/friendica/friendica/pull/10762). The same PR moves the follow revocation to the blocking feature, silently again.

This issue tracks the creation of an explicit control to revoke follows as suggested in https://github.com/friendica/friendica/pull/10751#issuecomment-927126951

It would be a separate button from the "Follow" button, depending on the current relationship status and the eventual support of the feature by the target network.

Potential locations of the button with my opinion for each:
- In the Contact list page, in a Contact row (already button-crowded)
- In the Contact page, in the Actions menu (confidential, appropriate given the edge use case)
- In the Profile vcard on the left (too prominent for a marginal feature)

### PR DESCRIPTION
**Pull request by MrPetovan**
Closes #10754
Blocks #10762

### CODE REVIEW

**Comment by MrPetovan** [comment]
Screenshot (47)

File: doc/Addons.md [hunk 1]
```diff
@@ -750,6 +773,13 @@ Here is a complete list of all hook callbacks with file locations (as of 24-Sep-
 
     Hook::callAll('logged_in', $a->user);
 
+### src/Core/Protocol.php
+
+    Hook::callAll('support_follow', $hook_data);
+    Hook::callAll('support_revoke_follow', $hook_data);
+    Hook::callAll('unfollow', $hook_data);
+	Kook::callAll('revoke_follow', $hook_data);
```
Comment (tobiasd) [inline doc/Addons.md]: This indentation looks wired ;-)

File: doc/de/Addons.md [hunk 2]
```diff
@@ -413,7 +412,14 @@ Eine komplette Liste aller Hook-Callbacks mit den zugehörigen Dateien (am 01-Ap
...
 
     Hook::callAll('logged_in', $a->user);
-    
+
+### src/Core/Protocol.php
+
+    Hook::callAll('support_follow', $hook_data);
+    Hook::callAll('support_revoke_follow', $hook_data);
+    Hook::callAll('unfollow', $hook_data);
+	Kook::callAll('revoke_follow', $hook_data);
```
Comment (tobiasd) [inline doc/de/Addons.md]: indentation here as well

File: doc/Addons.md (hunk 1 above)
Comment (MrPetovan) [reply doc/Addons.md]: Worse, it looks mixed!

**Pull request merged**